src/actions/DeepLinkingActions.ts

```diff
@@ -131,13 +131,13 @@ export async function handleLink(navigation: NavigationBase, dispatch: Dispatch,
     case 'promotion': {
       if (!state.account.accountReferralLoaded) return false
       const { installerId = '' } = link
-      dispatch(activatePromotion(installerId))
+      dispatch(activatePromotion(installerId)).catch(err => showError(err))
```

Comment:
The `handleLink` function should simply `await` these lines so the error flows out.

Then, we should put catch statements on the places where `handleLink` is being called:

- `SwapDetailsTile` will probably `showError`
- `retryPendingDeepLink` should log the problem and leave the link un-handled
- `launchDeepLink` should probably become `async`, so it can `await` the `handleLink`. This kicks the can up to the `launchDeepLink` callers.

Reply:
How is `handleLink` being called by `SwapDetailsTile`; I couldn't make that connection?

I implemented `retryPendingDeepLink` such that it catches internally; though correct me if I'm wrong, do we need a fail rate counter so we can avoid thrashing this retry routine?